beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/metadata/MetadataCleaner.java

```diff
@@ -19,9 +19,9 @@
 
 public interface MetadataCleaner {
 
-  void dropTable(HousekeepingMetadata housekeepingMetadata);
+  void dropTable(CleanerClient client, HousekeepingMetadata housekeepingMetadata);
```

Comment:
Summarising some discussions around this change - it's not great to have this coupling here and for the `CleanerClient` to be passed around to all the methods. Having this be an implementation detail of `MetadataCleaner` would be better but there are issues around where the client is created and the amount of logging noise it produces. I think we could solve this all in a nicer way by having some kind of HiveClientPool which manages connections and then the implementation of this could use that to manage connecting to Hive. I'm OK with approving this to move it forward but I'll set up a separate discussion to decide on whether creating a more generic HiveClientPool would be useful here and also potentially in some of our other projects.

Reply:
Thanks for summarising that, @massdosage. I was having the same conversation with Jay earlier this morning as I felt the API was not the best with the client being passed in with every interaction.

Reply:
That sounds good. The old client pool again :).

Reply:
we can also just add a factory here in the implementation of this class and create a new client for inside every method. A pool or what we have now is after all just an optimisation detail we might not need it. I think we actually established long time ago that creating a client is fast.

Reply:
Creating a new client inside every method isnt ideal. I tried it but theres two calls back to back `tableExists` and `dropTable` / `dropPartition` which create two new clients and this adds a lot to the logs which is annoying. As Adrian suggested we could just ignore those logs, but its good to know whats going on with the clients

Reply:
ah yes logging can be silenced. All good we can have. cache with a timeout. I think we're all single threaded here so that's fine.